nix/tullia.nix

```diff
@@ -0,0 +1,72 @@
+self: system:
+
+let
+  ciInputName = "GitHub event";
+in rec {
+  tasks = let
+    inherit (self.inputs.tullia) flakeOutputTasks taskSequence;
+
+    common = {
+      config,
+      ...
+    }: {
+      preset = {
+        # needed on top-level task to set runtime options
+        nix.enable = true;
+
+        github-ci = {
+          enable = config.actionRun.facts != {};
+          repo = "input-output-hk/plutus";
+          sha = config.preset.github-ci.lib.getRevision ciInputName null;
+        };
+      };
+    };
+
+    # the attribute name in `hydraJobs` for the current system
+    os = {
+      x86_64-linux = "linux";
+      x86_64-darwin = "macos";
+    }.${system};
+
+    mkHydraJobTask = flakeOutputTask: {
+      imports = [common flakeOutputTask];
+
+      # some hydra jobs run NixOS tests
+      env.NIX_CONFIG = ''
+        extra-system-features = kvm
+      '';
+
+      memory = 1024 * 8;
+      nomad.resources.cpu = 10000;
+    };
+    mkHydraJobTasks = __mapAttrs (_: mkHydraJobTask);
+
+    hydraJobTasks = mkHydraJobTasks (flakeOutputTasks ["hydraJobs" os] self);
+
+    ciTasks = taskSequence "ci/" hydraJobTasks (__attrNames hydraJobTasks);
+  in
+    ciTasks // {
+      "ci" = {lib, ...}: {
+        imports = [common];
+        after = [(lib.last (__attrNames ciTasks))];
+      };
+    };
+
+  actions = {
+    "plutus/ci" = {
+      task = "ci";
+      io = ''
+        let github = {
+          #input: "${ciInputName}"
+          #repo: "input-output-hk/plutus"
+        }
+
+        #lib.merge
+        #ios: [
+          #lib.io.github_push & github,
+          { #lib.io.github_pr, github, #target_default: false },
```

Comment:
What do these things do?

Reply:
This is the CUE expression that defines when to run the action. In particular, it uses tullia's library from `#lib` to build an expression that matches the JSON document that the GitHub webhook event sends on a push or PR.

Reply:
Worth explaining! or at least adding a link to where to find out!

Reply:
There are no docs yet :sweat_smile: